src/datasets/arrow_dataset.py

```diff
@@ -596,7 +609,7 @@ def update_metadata_with_features(table: Table, features: Features):
     return table
 
 
-def _check_table(table) -> Table:
+def _check_table(table: Union[pa.Table, Table]) -> Union[InMemoryTable, Table]:
```

Comment:
```suggestion
def _check_table(table: Union[pa.Table, Table]) -> Table:
```
InMemoryTable is a subclass of Table